### ISSUE TITLE
fix bug with name property of xugrid.UgridDataArray

### ISSUE THREAD

**Issue by DirkEilander** [Deltares/hydromt#273]
We get a `AttributeError: can't set attribute 'name'` on [model_mesh.py#L207](https://github.com/Deltares/hydromt/blob/main/hydromt/models/model_mesh.py#L207) and [conftest.py#L172](https://github.com/Deltares/hydromt/blob/main/tests/conftest.py#L172
)
see also https://github.com/Deltares/hydromt/actions/runs/3988723841/jobs/6840257511

Normal xarray.DataArray objects do have a name property and the used to have a name property, but if i'm reading the error message correctly this seems to have been removed in xugrid.

@Huite Does this issue ring a bell?
@hboisgon Not sure why we used the name property in the first place, do you know/remember?

**Comment by hboisgon** [comment]
Mesh works like the grid object so we add slowly variables (like elevation, landuse etc) to mesh dataset in set_mesh so we do need a name (similar to xarray.to_dataset). If name of Datarray is none I will get an error in xarray downstream anyway:

raise ValueError(
                "unable to convert unnamed DataArray to a "
                "Dataset without providing an explicit name"
            )

I can do a workaround by converting from xugrid to xarray objects and back or make name a required argument if instance is an array but to me it seems better to fix it in xugrid and for now put a pin in xugrid version?
@Huite what do you think?

**Comment by DirkEilander** [comment]
Does `data.rename(name)` still work? If `data` is a DataArray that should do the same in xarray. If we fix the xugrid version or add a workaround like this one it would be also good to flag this in an issue over at the xugrid github repos.

**Comment by Huite** [comment]
This doesn't ring a bell, and my test suite is still growing. I have been refactoring quite a bit recently, so I'll take a look tomorrow.

**Comment by Tjalling-dejong** [comment]
I fixed the bug. The problem was `data` was set to `data.name`. Should be `data.name = name`

**Comment by Tjalling-dejong** [comment]
Okay, still not fixed. Somehow tests passed locally, but not on github.

**Comment by Tjalling-dejong** [comment]
@DirkEilander data.rename(name) works both locally and on with github actions. 
I am still confused why locally the tests do not throw the AttributeError and with github actions it does.

**Comment by Huite** [comment]
Just testing this: pretty sure I broke this with the newest release of xugrid.

I overhauled the xarray wrapping process, it's much more powerful now: any xarray functions which results in a subset along a UGRID dimension would break the link with the UGRID topology before, but it creates a new topology now automatically.

Unfortunately, I might have broken some properties (the getters work, but the setters not so much judging from this issue).

@Tjalling-dejong: you likely have an older version locally. The Github action grabs the latest release.

**Comment by Huite** [comment]
Cross-ref: https://github.com/Deltares/xugrid/issues/51

My wrapping code doesn't take property setters into account. I'll try to fix this ASAP.

**Comment by Huite** [comment]
I've just pushed a commit and added a unit test on main.

I don't think the xarray properties implement `fdel` to delete properties, but this seems to do the trick:

```python
        wrapped = property(
            fget=wraps_xarray(prop.__get__),
            fset=wraps_xarray(prop.__set__),
            doc=prop.__doc__,
        )
        target_class_dict[name] = wrapped
```

I'll make a new release soon, but I'm hoping to fix another bug or two.

**Comment by hboisgon** [comment]
Hi @Huite thank you so much for looking into it! Then we just have to wait for the release on our side, this also is better than some workaround :)

**Comment by DirkEilander** [comment]
this is fixed in #217
